compiler/passes/scopeResolve.cpp

```diff
@@ -2042,6 +2114,73 @@ static void buildBreadthFirstModuleList(Vec<ModuleSymbol*>* modules,
   }
 }
 
+// Returns true if this module is capable of being used or traversed as part of
+// an access in the provided scope, false if the module is private and the
+// scope is not its direct parent or a sibling module when the parent of both
+// had a use of it.
+bool ModuleSymbol::isVisible(BaseAST* scope) {
+  if (hasFlag(FLAG_PRIVATE)) {
+    BaseAST* parentScope = getScope(defPoint);
+    INT_ASSERT(parentScope != NULL); // Should be true, given we found this
+    // module symbol.
+
+    if (scope == parentScope) {
+      // The scope provided is the scope in which we are defined
+      return true;
+    } else {
+      // Well, the scope is not our immediate parent.  It could be a subscope
+      // of our parent scope, though, so we need to walk up scopes until we
+      // either find our parent scope (in which case, we're visible if it "use"s
+      // us) or we run out of scope to check against (in which case we are most
+      // certainly *not* visible)
+      BaseAST* searchScope = getScope(scope);
+      bool passedModule = false;
```

Comment:
Hmmm, good point!  I'll follow your suggestions and let you know when I've committed an updated version to make sure it is clear.